### PR TITLE
Okta: get user info into store

### PR DESCRIPTION
**Pull request by fabio-silva**
- Cleaned login / logout unused actions
- `useUserInfo` hook takes care of calling Okta's endpoint and fill in the store the first time

### CODE REVIEW

File: src/core/hooks/useUserInfo.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect } from 'react';
+import { useOktaAuth } from '@okta/okta-react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AuthState, UpdateProfilePayload } from 'store/types';
+import { RequestError } from 'core/api/types';
+import { authGetProfileAction, authUpdateProfileAction, getAuth } from '../../store/auth';
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:6]:
```suggestion
import { authGetProfileAction, authUpdateProfileAction, getAuth } from 'store/auth';
```
nit

File: src/core/hooks/useUserInfo.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export const useUserInfo = (): [AuthState, (payload: UpdateProfilePayload) => void] => {
+  const { oktaAuth } = useOktaAuth();
+  const dispatch = useDispatch();
+  const user = useSelector(getAuth);
+  const { freshData } = user;
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:12]: `freshData` reads somewhat weird )
Is it us or Okta who came up with that prop name?

File: src/store/auth/auth.reducer.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
 import { createAsyncAction, ActionType, getType } from 'typesafe-actions';
-import { AuthState, LoginPayload, LogoutPayload, UpdateProfilePayload } from 'store/types';
+import { AuthState, UpdateProfilePayload } from 'store/types';
 import { RequestError } from 'core/api/types';
 
 const DEFAULT_STATE: AuthState = {
   authenticated: false,
+  freshData: true,
```
Comment (atymchuk) [inline src/store/auth/auth.reducer.ts:7]: How can `freshData` be true by default? Not sure it's even needed )

File: src/core/hooks/useUserInfo.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  }, [dispatch]);
+
+  useEffect(() => {
+    const getInfo = async() => {
+
+      if (freshData) {
+        dispatch(authGetProfileAction.request());
+
+        try {
+          const { email: oktaEmail, family_name, given_name } = await oktaAuth.getUser();
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:25]:
```suggestion
          const { email, family_name, given_name } = await oktaAuth.getUser();
```

File: src/core/hooks/useUserInfo.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ 
+          dispatch(authGetProfileAction.success({
+            email: oktaEmail,
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:28]:
```suggestion
            email,
```

File: src/core/hooks/useUserInfo.ts (hunk 4 above)
Comment (atymchuk) [reply src/core/hooks/useUserInfo.ts:25]: No need to be renamed, the email is not Okta's.

**Review by atymchuk — CHANGES_REQUESTED** [review]

**Review by nicolalamacchia — DISMISSED** [review]
just found some nits @atymchuk already pointed to

File: src/core/hooks/useUserInfo.ts (hunk 1 above)
Comment (fabio-silva) [reply src/core/hooks/useUserInfo.ts]: 👍

File: src/core/hooks/useUserInfo.ts (hunk 2 above)
Comment (fabio-silva) [reply src/core/hooks/useUserInfo.ts]: removed flag

File: src/core/hooks/useUserInfo.ts (hunk 4 above)
Comment (fabio-silva) [reply src/core/hooks/useUserInfo.ts]: I renamed this var. It must have another name, as we already got `email` some lines above (`const { email } = user;`)

File: src/core/hooks/useUserInfo.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  }, [dispatch]);
+
+  useEffect(() => {
+    const getInfo = async() => {
+
+      if (!email) {
+        dispatch(authGetProfileAction.request());
+
+        try {
+          const { email: oktaUserEmail, family_name, given_name } = await oktaAuth.getUser();
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:25]:
```suggestion
          const { email, family_name, given_name } = await oktaAuth.getUser();
```

File: src/core/hooks/useUserInfo.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+ 
+          dispatch(authGetProfileAction.success({
+            email: oktaUserEmail,
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:28]:
```suggestion
            email,
```

File: src/core/hooks/useUserInfo.ts (hunk 6 above)
Comment (atymchuk) [reply src/core/hooks/useUserInfo.ts:25]: it's a nit, but why do we have to attach the word "okta" to the properties? It is a user's email after all

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: src/core/hooks/useUserInfo.ts (hunk 6 above)
Comment (fabio-silva) [reply src/core/hooks/useUserInfo.ts]: good one 👍

File: src/core/hooks/useUserInfo.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        } catch (e) {
+          dispatch(authGetProfileAction.failure(e as RequestError));
+        }
+      }
+    };
+
+    getInfo();
+  }, [oktaAuth, dispatch, email]);
+
+  return [{ ...user }, setUser];
```
Comment (atymchuk) [inline src/core/hooks/useUserInfo.ts:41]: No need to copy imho, it will always return a new reference whenever things change in the user's state

**Review by atymchuk — APPROVED** [review]

**Pull request closed without merging**